docs/extraction.md

```diff
@@ -32,44 +90,73 @@ classes:
 
 Configure the extraction model and prompting strategy:
 
+#### Configuration for Agentic Extraction (Recommended)
+
 ```yaml
 extraction:
-  # Model selection and parameters
-  model: us.amazon.nova-pro-v1:0
-  temperature: 0.0
+  # Enable agentic extraction for better accuracy
+  agentic:
+    enabled: true # Turn on for production use
+
+  # Model selection - must support tool use for agentic
+  model: anthropic.claude-3-5-sonnet-20241022-v2:0 # Recommended for best results
+  # Alternative models:
+  # - anthropic.claude-3-7-sonnet-20250219-v1:0  # Latest Sonnet with enhanced capabilities
+  # - us.amazon.nova-pro-v1:0  # AWS native option
+  temperature: 0.0 # Keep low for consistency
   top_p: 0.1
   top_k: 5
   max_tokens: 4096
-  
+
   # Prompts for extraction
   system_prompt: |
     You are an expert in extracting structured information from documents.
     Focus on accuracy in identifying key fields based on their descriptions.
     For each field, look for both the field label and the associated value.
     Pay attention to formatting patterns common in business documents.
     When a field is not present, indicate this explicitly rather than guessing.
-    
+
   task_prompt: |
```

Comment:
Can you explain how the system_prompt and task_prompts apply to the agentic extraction..  Does the agent use the same prompt as 'traditional' method?